tools/workflow/clean_aws_resources.go

```diff
@@ -97,17 +105,172 @@ func terminateEc2Instances() {
+			if !time.Now().UTC().Add(pastDayDeleteCalculation).After(*asg.CreatedTime) {
+				continue
+			}
+
+			deleteAutoScalingGroupInput := &autoscaling.DeleteAutoScalingGroupInput{
+				AutoScalingGroupName: asg.AutoScalingGroupName,
+				ForceDelete:          aws.Bool(true),
+			}
+
+			_, err = autoscalingclient.DeleteAutoScalingGroup(deleteAutoScalingGroupInput)
+
+			if err != nil {
+				log.Fatalf("Failed to delete asg %s because of %v", *asg.AutoScalingGroupName, err)
+			}
+
+			deleteLaunchConfigurationInput := &autoscaling.DeleteLaunchConfigurationInput{
+				LaunchConfigurationName: asg.LaunchConfigurationName,
+			}
+
+			_, err = autoscalingclient.DeleteLaunchConfiguration(deleteLaunchConfigurationInput)
+
+			if err != nil {
+				log.Fatalf("Failed to delete lc of asg %s  because of %v", *asg.AutoScalingGroupName, err)
+			}
+
+			log.Printf("Delete asg %s successfully", *asg.AutoScalingGroupName)
+		}
+
+		if describeAutoScalingOutputs.NextToken == nil {
+			break
+		}
+
+		nextToken = describeAutoScalingOutputs.NextToken
+	}
+}
+
+func destroyECSLaunchConfiguration() {
+	//Set up aws go sdk ec2 client
+	testSession, err := session.NewSession()
+
+	if err != nil {
+		log.Fatalf("Error creating session %v", err)
+	}
+
+	ec2client := ec2.New(testSession)
+	autoscalingclient := autoscaling.New(testSession)
+
+	//Allow to load all the launch configurations since the default respond is paginated launch configurations.
+	//Look into the documentations and read the starting-token for more details
+	//Documentation: https://docs.aws.amazon.com/cli/latest/reference/autoscaling/describe-launch-configurations.html#options
+	var nextToken *string
+
+	for {
+		//Launch Configuration Go SDK currently does not support filter tag or filter wildcard. Only supports with matching name
+		//Documentation: https://github.com/aws/aws-sdk-go/blob/02266ed24221ac21bb37d6ac614d1ced95407556/service/autoscaling/api.go#L9577-L9593
+		describeLaunchConfigurationInputs := &autoscaling.DescribeLaunchConfigurationsInput{NextToken: nextToken}
+		describeLaunchConfigurationOutputs, err := autoscalingclient.DescribeLaunchConfigurations(describeLaunchConfigurationInputs)
+
+		if err != nil {
+			log.Fatalf("Failed to get metadata from launch configuration because of %v", err)
+		}
+
+		for _, lc := range describeLaunchConfigurationOutputs.LaunchConfigurations {
+
+			//Skipping lc that does not contain cluster-aoc-testing string (relating to aws-otel-test-framework)
+			if filterLcNameResult := strings.Contains(*lc.LaunchConfigurationName, containLcName); !filterLcNameResult {
+				continue
+			}
+
+			//Skipping lc that does not older than 5 days
+			if !time.Now().UTC().Add(pastDayDeleteCalculation).After(*lc.CreatedTime) {
+				continue
+			}
+
+			//Split the instance profile to get the testing ID associate with the testing framework
+			splitLcIamInstanceProfile := strings.Split(*lc.IamInstanceProfile, "-")
+
+			if len(splitLcIamInstanceProfile) != 6 {
```

Comment:
6 feels arbitrary. Make it a named const.

Reply:
Reasonable. Changed to ```requiredIAMInstanceProfileLength``` in newest commit. About splitting, agree. Still finding a boilerpoint to do it. Moreover, one problem to address when splitting the file is: should we still used go language and maintaining it dependencies (same as tool linters) or should we use bash? Personally, I would prefer bash but going to far to turn back to use bash now.

Reply:
It doesn't hurt to still use go. If we do split it up, it shouldn't be in this PR.